### ISSUE TITLE
Add File History

### ISSUE THREAD

**Issue by moshihe** [zed-industries/community#1824]
### Check for existing issues

- [X] Completed

### Describe the feature

Last week I wrote a lot of code using the zed editor. But I didn't use git to save it. Due to my own mishandling, and I don't understand what I did, I lost the code. In jetbrain products like goland I can easily find the code by using the history function.

With zed, I've also encountered that when saving with command + s, if I do it too fast, I get a popup box saying that the file was overwritten. I'm not sure if this is a bug or not, he didn't affect my use.

It was my request, not a flaw in the product. I hope zed gets better and better.

**Comment by JosephTLyons** [comment]
Sorry to hear that you lost code, @moshihe. Zed should be doing some sort of incremental backup / file history, like JetBrains does.

**Comment by moshihe** [comment]
Thanks for the reply, the zed editor is great. The history feature may be a bit heavy for an editor. I'm talking about the History feature for file changes, not autosave.

Anyway, thanks for developing awesome products.

**Comment by moshihe** [comment]
Yes. Just like the picture below. It's a Goland from jetbrains .

![image](https://github.com/zed-industries/community/assets/136976791/803deaf9-0a87-420f-9ffe-68644cb94b42)
